components/aws/sagemaker/batch_transform/src/sagemaker_transform_spec.py

```diff
@@ -0,0 +1,218 @@
+    SageMakerComponentInputValidator as InputValidator,
+    SageMakerComponentOutputValidator as OutputValidator,
+)
+
+
+@dataclass(frozen=True)
+class SageMakerTransformInputs(SageMakerComponentCommonInputs):
+    """Defines the set of inputs for the transform component."""
+
+    job_name: Input
+    model_name: Input
+    max_concurrent: Input
+    max_payload: Input
+    batch_strategy: Input
+    environment: Input
+    input_location: Input
+    data_type: Input
+    content_type: Input
+    split_type: Input
+    compression_type: Input
+    output_location: Input
+    accept: Input
+    assemble_with: Input
+    output_encryption_key: Input
+    input_filter: Input
+    output_filter: Input
+    join_source: Input
+    instance_type: Input
+    instance_count: Input
+    resource_encryption_key: Input
+
+
+@dataclass
+class SageMakerTransformOutputs(SageMakerComponentBaseOutputs):
+    """Defines the set of outputs for the transform component."""
+
+    output_location: Output
+
+
+class SageMakerTransformSpec(
+    SageMakerComponentSpec[SageMakerTransformInputs, SageMakerTransformOutputs]
+):
+    INPUTS: SageMakerTransformInputs = SageMakerTransformInputs(
+        job_name=InputValidator(
+            input_type=str,
+            required=False,
+            description="The name of the transform job.",
+            default="",
+        ),
+        model_name=InputValidator(
+            input_type=str,
+            required=True,
+            description="The name of the model that you want to use for the transform job.",
+        ),
+        max_concurrent=InputValidator(
+            input_type=int,
+            required=False,
+            description="The maximum number of parallel requests that can be sent to each instance in a transform job.",
+            default="0",
+        ),
+        max_payload=InputValidator(
+            input_type=int,
+            required=False,
+            description="The maximum allowed size of the payload, in MB.",
+            default="6",
+        ),
+        batch_strategy=InputValidator(
+            input_type=str,
+            choices=["MultiRecord", "SingleRecord", ""],
+            required=False,
+            description="The number of records to include in a mini-batch for an HTTP inference request.",
+            default="",
+        ),
+        environment=InputValidator(
+            input_type=SpecValidators.yaml_or_json_dict,
+            required=False,
+            description="The dictionary of the environment variables to set in the Docker container. Up to 16 key-value entries in the map.",
+            default={},
+        ),
+        input_location=InputValidator(
+            input_type=str,
+            required=True,
+            description="The S3 location of the data source that is associated with a channel.",
+        ),
+        data_type=InputValidator(
+            input_type=str,
+            choices=["ManifestFile", "S3Prefix", "AugmentedManifestFile", ""],
```

Comment:
strange that we have empty string as option for some of the inputs

Reply:
From [this StackOverflow post](https://stackoverflow.com/questions/40324356/python-argparse-choices-with-a-default-choice/40324463) it appears that the default needs to be a choice?